ush/python/pygfs/jedi/jedi.py

```diff
@@ -216,18 +218,45 @@ def get_bias_dict(self, task_config: AttrDict) -> Dict[str, Any]:
                 obfile = ob['obs bias']['input file']
                 obdir = os.path.dirname(obfile)
                 basename = os.path.basename(obfile)
-                prefix = '.'.join(basename.split('.')[:-2])
-                for file in ['satbias.nc', 'satbias_cov.nc', 'tlapse.txt']:
-                    bfile = f"{prefix}.{file}"
-                    copylist.append([os.path.join(task_config.COM_ATMOS_ANALYSIS_PREV, bfile), os.path.join(obdir, bfile)])
-                    # TODO: Why is this specific to ATMOS?
+                prefix = '.'.join(basename.split('.')[:-3])
+                bfile = f"{prefix}.{bias_file}"
+                tar_file = os.path.join(obdir, bfile)
+                copylist.append([os.path.join(task_config.VarBcDir, bfile), tar_file])
+                break
 
         bias_dict = {
             'mkdir': [os.path.join(task_config.DATA, 'bc')],
             'copy': copylist
         }
+
         return bias_dict
 
+    @logit(logger)
+    def extract_tar(self, task_config: AttrDict, tar_dict) -> Dict[str, Any]:
+        """Extract files from list of tarfiles
+
+        This method extract bias correction files from tarball(s)
+
+        Parameters
+        ----------
+        task_config: AttrDict
+            Attribute-dictionary of all configuration variables associated with a GDAS task.
+        tar_dict
+            a dictionary containing the list of tar files
+
+        Returns
+        ----------
+        None
+        """
+
+        # extract bias correction files from tar file
+        for tar_file in tar_dict['copy']:
+            if ".tar" in tar_file[1]:
+                with tarfile.open(tar_file[1], "r") as tarball:
+                    tarball.extractall(path=os.path.join(task_config.DATA, 'obs'))
+                    logger.info(f"Extract {tarball.getnames()}")
+        tarball.close()
```

Comment:
This is a good place for a `try-except` 
```python3
try:
    with tarfile.open(tar_file[1], 'r') as tarball:
        tarball.extractall(path=os.path.join(task_config.DATA, 'obs'))
        logger.info(f"Extract {tarball.getnames()}")
except tarfile.ReadError as err:
    if tarfile.is_tarfile(tarfile[1]):
        logger.error(f"FATAL ERROR: {tarfile[1]} is could not be read")
        raise tarfile.ReadError(f"FATAL ERROR: unable to read {tarfile[1]}")
    else:
        logger.info()
except tarfile.ExtractError as err:
    logger.exception(f"FATAL ERROR: unable to extract from file")
    raise tarfile.ExtractError("FATAL ERROR ...")
```
Raising the right kind of error will help debugging and finding the root cause sooner.

Reply:
Good suggestion.  The recommended refactoring accepted and committed at 0ab7bdd6.   [tarfile documentation](https://docs.python.org/3/library/tarfile.html) lists many error conditions.  Are there other types of errors we should trap?

Reply:
@aerorahul , please resolve if [0ab7bdd](https://github.com/NOAA-EMC/global-workflow/commit/0ab7bdd6083051f5cb9b9bc687d2a4bbbe5e2e34) is acceptable.